src/core_processor/session/session_manager.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import time
+import traceback
+from pathlib import Path
+
+import cv2
+
+from src.cameras.multicam_manager.cv_camera_manager import OpenCVCameraManager
+from src.cameras.persistence.video_writer.save_options import SaveOptions
+from src.core_processor.camera_calibration.camera_calibrator import CameraCalibrator
+from src.core_processor.fps.fps_counter import FPSCamCounter
+from src.core_processor.show_cam_window import show_cam_window
+
+logger = logging.getLogger(__name__)
+
+
+class SessionManager:
+    def __init__(self,
+                 opencv_camera_manager: OpenCVCameraManager = OpenCVCameraManager(),
+                 calibrate_cameras: bool = True
+                 ):
+        self._open_cv_camera_manager = opencv_camera_manager
+        self.calibrate_cameras = calibrate_cameras
+        if self.calibrate_cameras:
+            self.camera_calibrator = CameraCalibrator()
+        self._start_time = time.time()
+
+    async def process_by_cam_id(self, webcam_id: str, cb):
+        with self._open_cv_camera_manager.start_capture_session_single_cam(
+                webcam_id=webcam_id,
+        ) as session_obj:
+            fps_manager = FPSCamCounter(self._open_cv_camera_manager.available_webcam_ids)
+            fps_manager.start_all()
+            cv_cam = session_obj.cv_cam
+            writer = session_obj.writer
+            try:
+                while True:
+                    if not cv_cam.is_capturing_frames:
```

Comment:
this could be a timing issue - we could call start within the cv cam, but not be capturing frames yet for a camera and the entire method would exit. i think this should be a `continue` with a debug log